artifactory/services/utils/replication.go

```diff
@@ -1,14 +1,11 @@
 package utils
 
-import "encoding/json"
-
-type ReplicationBody struct {
+type replicationBody struct {
```

Comment:
One downside of this change to have two new structs of `GetReplicationBody` and `UpdateReplicationBody` rather than overriding the JSONUnmarshal function is that this change is no longer backwards compatible.

Even if I keep choose to keep `replicationBody` exposed, the function specs in [artifactory/services/createreplication.go](https://github.com/jfrog/jfrog-client-go/pull/524/commits/e7c1168ef582b47998ad992e2ba1b7aecc45a366#diff-bc8309bdb397edd2769ac91efbcfc86a9935ca146a9c83bcee9666440a306edb) and [artifactory/services/getreplication.go](https://github.com/jfrog/jfrog-client-go/pull/524/commits/e7c1168ef582b47998ad992e2ba1b7aecc45a366#diff-0f25ef08a33a2041ce11c417ad456871009fd7bcd6b427719f678e5fe06113d4) will be changed which are backward incompatible changes.

Reply:
@Guent4 -
We're mostly okay with breaking compatibility following APO changes. If a public API changed, we make sure to note it in the release notes.